charts/nri-prometheus/templates/configmap.yaml

```
@@ -11,3 +11,8 @@ data:
 {{- if .Values.config }}
 {{ toYaml .Values.config | indent 4 -}}
 {{ end }}
+{{- if .Values.guidedInstall }}
+{{ toYaml .Values.guided_install_transformations | indent 4 -}}
+{{ else }}
+{{ toYaml .Values.custom_transformations | indent 4 -}}
+{{ end }}
```

I'm not much of a fan of the idea of merging keys in the yaml this way. For example, it would be possible for an unknowledgeable user to add a `transformations` section to `.Values.config`, causing a yaml syntax error.

Since with the guided install it's very uncommon to have a custom values.yml file, what do you think about creating a different template, containing a whole config file for guided install, and leaving this as it is?

Something like:

```suggestion
{{- if .Values.guidedInstall }}
{{ include "guidedInstallConfig" . | indent 4}}
{{- else if .Values.config }}
{{ toYaml .Values.config | indent 4 -}}
{{- end }}
```

This way it would be very hard to accidentally mix configurations, and the default guided install config file could be on a separate template rather than in the values. IMO this would be better since it's not intended to be modified/overridden by the user.

---

@roobre I made some modifications to this in the latest update.  

- Created a `static` directory which contains the full YAML for the Guided Install config
- If `guidedInstall` is `true` in `values.yaml` then `static/guidedinstalldefaults.yaml` will be loaded into the POMI `config.yaml` ConfigMap
- If `guidedInstall is `false` in `values.yaml` then the `config:` section will be used.

Overall, I think this will be cleaner and we can easily edit the Guided Install defaults if we decide to change them in the static file.